src/test/scala/io/iohk/ethereum/blockchain/sync/BlockBroadcastSpec.scala

```diff
@@ -15,42 +15,60 @@ import org.scalatest.{FlatSpec, Matchers}
 
 class BlockBroadcastSpec extends FlatSpec with Matchers  {
 
-  it should "send a new block only when it is not known by the peer (known by comparing total difficulties)" in new TestSetup {
+  it should "send a new block when it is not known by the peer (known by comparing total difficulties)" in new TestSetup {
     //given
-    //Block that shouldn't be sent as it's number and total difficulty is lower than known by peer
-    val firstHeader: BlockHeader = baseBlockHeader.copy(number = initialPeerInfo.maxBlockNumber - 2)
-    val firstBlock = NewBlock(Block(firstHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty - 2)
-
     //Block that should be sent as it's total difficulty is higher than known by peer
-    val secondHeader: BlockHeader = baseBlockHeader.copy(number = initialPeerInfo.maxBlockNumber - 3)
-    val secondBlockNewHashes = NewBlockHashes(Seq(PV62.BlockHash(secondHeader.hash, secondHeader.number)))
-    val secondBlock = NewBlock(Block(secondHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty + 2)
+    val blockHeader: BlockHeader = baseBlockHeader.copy(number = initialPeerInfo.maxBlockNumber - 3)
+    val newBlockNewHashes = NewBlockHashes(Seq(PV62.BlockHash(blockHeader.hash, blockHeader.number)))
+    val newBlock = NewBlock(Block(blockHeader, BlockBody(Nil, Nil)), initialPeerInfo.totalDifficulty + 2)
 
     //when
-    blockBroadcast.broadcastBlocks(Seq(firstBlock, secondBlock), Map(peer -> initialPeerInfo))
+    blockBroadcast.broadcastBlock(newBlock, Map(peer -> initialPeerInfo))
 
     //then
-    etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(secondBlock, peer.id))
-    etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(secondBlockNewHashes, peer.id))
+    etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(newBlock, peer.id))
+    etcPeerManagerProbe.expectMsg(EtcPeerManagerActor.SendMessage(newBlockNewHashes, peer.id))
     etcPeerManagerProbe.expectNoMsg()
   }
 
-  it should "send a new block only when it is not known by the peer (known by comparing max block number)" in new TestSetup {
+  it should "not send a new block when it is not known by the peer (known by comparing total difficulties)" in new TestSetup {
```

Comment:
`"send a new block when it is not known by the peer (known by comparing total difficulties)"`
`"not send a new block when it is not known by the peer (known by comparing total difficulties)"`

Having a cake and eating it too? 😉

Reply:
You are right, I missed this when splitting the test `"send a new block only when it is not known by the peer (known by comparing total difficulties)"` into one for when the block is broadcasted and one when it's not.
This was changed due to changing `BlockBroadcast.broadcastBlock` receiving only a `NewBlock` instead of a `Seq[NewBlock]`.